src/server/common.h

```diff
@@ -295,7 +295,7 @@ class Context : protected Cancellation {
 };
 
 struct ScanOpts {
-  std::string_view pattern;
+  std::optional<std::string_view> pattern;
```

Comment:
so there can be an empty pattern or are those just cosmetical changes?

Reply:
yes, there can be an empty pattern. I added a test for this as well